src/DPD-BASIC/pair_dpd.h

```diff
@@ -28,11 +28,11 @@ class PairDPD : public Pair {
  public:
   PairDPD(class LAMMPS *);
   ~PairDPD() override;
-  void compute(int, int) override;
+  virtual void compute(int, int) override;
```

Comment:
Why add "virtual" here? The "override" should be sufficient.

Reply:
Habit mostly, I can remove all the virtuals in the DPD dir, I typically use virtual if someone builds on top of it.

Reply:
As far as I understood the standard, functions don't lose their polymorphic status when "override" is used (they do without). So "virtual" is needed for the first definition and then "override" for all following unless you use "final" to prohibit further overrides.

Reply:
That is the way I understand it as well.  I will remove the added virtuals as that is the style of LAMMPS as of a bit ago.

Reply:
Perhaps some background info is helpful: we've run the code through clang-tidy to modernize some common constructs and would like to keep all new code consistent with that, so we don't have to do the clang-tidy processing all the time. :wink:

Reply:
>So "virtual" is needed for the first definition and then "override" for all following unless you use "final" to prohibit further overrides.

I wasn't aware of this--will keep this in mind for future code as well.

Reply:
I pushed code without the virtual where it isn't needed, it is still needed in allocate cuz that is the start of the chain.